ush/gfs/task/analysis.py

```diff
@@ -0,0 +1,411 @@
+        obs_list_dict['OBS_PREFIX'] = os.environ['OPREFIX']
+        obs_list_dict = ufsda.yamltools.parse_config(obs_list_dict)
+        # get observers from master dictionary
+        observers = obs_list_dict['observers']
+        obs_dict = {}
+        for ob in observers:
+            obfile = ob['obs space']['obsdatain']['engine']['obsfile']
+            basename = os.path.basename(obfile)
+            obs_dict[os.path.join(os.environ['COMIN_OBS'], basename)] = obfile
+        return obs_dict
+
+    def get_staticb_dict(self):
+        """
+        get dictionary of staticb files to copy/use
+        based on the specified configuration.
+        """
+        import ufsda # temporary until this is in workflow
+        import yaml
+        with open(self.berror_yaml, 'r') as yamlopen:
+            berror_yaml_dict = yaml.safe_load(yamlopen)
+
+    def stage_obs(self, filedict):
+        logging.info('Staging observations')
+        self.stage(filedict, skip_missing=True)
+        logging.info('Finished staging observations')
+
+    def stage_bkg(self, filedict):
+        logging.info('Staging model backgrounds')
+        self.stage(filedict)
+        logging.info('Finished staging backgrounds')
+
+    def stage_fix(self, filedict):
+        logging.info('Staging fix files')
+        self.stage(filedict)
+        logging.info('Finished staging fix files')
+
+    def stage_berror(self, filedict):
+        logging.info('Staging staticb files')
+        self.stage(filedict)
+        logging.info('Finished staging staticb files')
+
+    def stage(self, filedict, skip_missing=False):
+        for src, dest in filedict.items():
+            destdir = os.path.dirname(dest)
+            if not os.path.exists(destdir):
+                logging.info(f'{destdir} does not exist, creating directory.')
+                os.makedirs(destdir)
+            if os.path.exists(dest):
+                os.remove(dest)
+            if skip_missing:
+                if not os.path.exists(src):
+                    logging.warning(f'{src} does not exist. Will not copy.')
+                    continue
+            logging.info(f'Copying {src} to {dest}')
+            shutil.copyfile(src, dest)
+
+    def generate_yaml(self, extra_config={}):
+        """
+        Use existing tools to generate YAML from config and template
+        """
+        import copy
+        import ufsda # temporary until this is in workflow
+        output_yaml = f'{self.taskname}_{self.cdate}.yaml'
+        output_yaml_path = os.path.join(self.datadir, output_yaml)
+        template = self.yamltemplate
+        logging.info(f'Generating analysis YAML using {template}')
+        full_config = copy.deepcopy(self.config)
+        full_config.update(extra_config)
+        ufsda.yamltools.genYAML(full_config, template=template, output=output_yaml_path)
+        logging.info(f'Wrote YAML file to {output_yaml_path}')
+
+
+class AerosolAnalysis(Analysis):
```

Comment:
This second class should probably be in its own file.

Reply:
yep, I'll separate them out, thanks